Add ScalarWaveWorldtube library

nikwit opened the pull request:
This adds the worldtube library. I think it is best to add it directly to `CurvedScalarWave` since it only works with this system.

nikwit committed:
Add worldtube library

nilsvu commented on src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:
```diff
@@ -0,0 +1,14 @@
+
+#include "Options/Options.hpp"
+
+namespace CurvedScalarWave::Worldtube {
```
Add dox to this namespace please. For more extensive docs you can add a `Worldtube.hpp` file that only documents the namespaces.

nilsvu commented on src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:
```diff
@@ -0,0 +1,14 @@
+#include "Options/Options.hpp"
+
+namespace CurvedScalarWave::Worldtube {
+namespace OptionTags {
```
Add dox here too please so the tags show up in Doxygen. Can be just one line.

nikwit commented:
@nilsvu thanks. I added some initial docs and will add more specific docs once I merge the functionalities so I can reference them. I squashed directly since there was not much left of the original commit.

nilsvu reviewed (dismissed):
Nice description of the scheme :)

nilsvu commented on src/Evolution/Systems/CurvedScalarWave/Worldtube/CMakeLists.txt:
```diff
@@ -0,0 +1,26 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+set(LIBRARY Worldtube)
```
Oh maybe name this lib something like `CswWorldtube` or `ScalarWaveWorldtube`, because you want another GH worldtube lib later?

nilsdeppe replied:
^ agreed, and to distinguish it from CCE more, since CCE also uses worldtubes!
